mscore/masterpalette.cpp

```diff
@@ -263,5 +265,132 @@ void MasterPalette::changeEvent(QEvent *event)
+                  selectItem(treeWidget->topLevelItem(idx+1)->text(0));
+                  } 
+            else if (idx == treeWidget->topLevelItemCount() - 1) {
+                  if (!currentItem->isExpanded())
+                        selectItem(treeWidget->topLevelItem(0)->text(0));
+                  else
+                        treeWidget->setCurrentItem(symbolItem->child(0));
+                  }
+            else {
+                  int i;
+                  for (i = 0; i < symbolItem->childCount(); i++) {
+                        if (symbolItem->child(i) == currentItem)
+                              break;
+                        }
+                  if (i < symbolItem->childCount()-1) {
+                        treeWidget->setCurrentItem(symbolItem->child(i+1));
+                        }
+                  else {
+                        selectItem(treeWidget->topLevelItem(0)->text(0));
+                        }
+                  }
+            }
+      else if (event->key() == Qt::Key_Up) {
+            if (idx > 0)
+                  selectItem(treeWidget->topLevelItem(idx-1)->text(0));
+            else if (idx == 0)
+                  selectItem(treeWidget->topLevelItem(treeWidget->topLevelItemCount() - 1)->text(0));
+            else {
+                  int i;
+                  for (i = 0; i < symbolItem->childCount(); i++) {
+                        if (symbolItem->child(i) == currentItem)
+                              break;
+                        }
+                  if (i > 0) {
+                        treeWidget->setCurrentItem(symbolItem->child(i-1));
+                        }
+                  else {
+                        selectItem(treeWidget->topLevelItem(treeWidget->topLevelItemCount() - 1)->text(0));
+                        }
+                }
+            }
+      else if (event->key() == Qt::Key_Right) {
+            QWidget* w = stack->currentWidget();
+            QWidget* ch = w->childAt(1, 1);
+            if (ch) {
+                  Palette* chp = static_cast<Palette*>(ch);
+                  chp->setSelected(0);
+                  chp->update();
+                  chp->setFocus();
+                  }
+            else {
+                  if (stack->currentIndex() == 1) {
+                        KeyEditor* k = static_cast<KeyEditor*>(w);
+                        Palette* p = k->getPalette();
+                        p->setSelected(0);
+                        p->update();
+                        p->setFocus();
+                        }
+                  else if (stack->currentIndex() == 2) {
+                        TimeDialog* t = static_cast<TimeDialog*>(w);
+                        Palette* p = t->getPalette();
+                        p->setSelected(0);
+                        p->update();
+                        p->setFocus();
+                        }
+                  else if (stack->currentIndex() >= 24) {
```

Comment:
What is 24 ?

Reply:
Looks like this is intended to be the Symbols section.  But there's got to be a better way of figuring out which section we are in that depending on the index number.  This will break if we add new sections to the master palette, or reorder them.  Same with special casing 1 & 2 for key and time signatures.